src/main/kotlin/org/rust/ide/intentions/CreateFunctionIntention.kt

```diff
@@ -37,16 +38,25 @@ class CreateFunctionIntention : RsElementBaseIntentionAction<CreateFunctionInten
 
     sealed class Context(val name: String, val callElement: PsiElement) {
         abstract val visibility: String
+        open val isAsync: Boolean = callElement.parentsOfType<RsBlock>().firstOrNull().let { block ->
+            when {
+                block?.parentsOfType<RsBlockExpr>()?.any { it.isAsync } == true -> true
+                block?.parentOfType<RsFunction>()?.isAsync == true -> true
+                else -> false
+            }
+        }
         abstract val arguments: RsValueArgumentList
         abstract val returnType: Ty?
         open val implItem: RsImplItem? = null
 
-        open class Function(val callExpr: RsCallExpr, name: String, val module: RsMod) : Context(name, callExpr) {
+        open class Function(callExpr: RsCallExpr, name: String, val module: RsMod) : Context(name, callExpr) {
             override val visibility: String = when {
                 callExpr.containingCrate != module.containingCrate -> "pub "
                 callExpr.containingMod != module -> "pub(crate) "
                 else -> ""
             }
+            override val isAsync: Boolean = super.isAsync
+                && callExpr.parentOfType<RsDotExpr>()?.fieldLookup?.isAsync == true
```

Comment:
This check should be more precise, to only check the direct parent.
Otherwise e.g. this:
```rust
async fn foo() {
    baz(/*caret*/bar()).await;
}
async fn baz(a: u32) {}
```
will produce
```rust
async fn foo() {
    baz(bar()).await;
}
async fn bar() -> u32 {
    unimplemented!()
}
async fn baz(a: u32) {}
```
Which is probably not what the user wanted.